feat(cli): `generate`, add `--no-hint` argument

nickcarnival opened the pull request:
This PR aims to remove some of the bloat that's output everytime you run `prisma generate` by adding a `--no-hint` flag to the `generate` command:

Closes #22513



```sh
prisma generate --no-hint
```

![Screenshot 2024-05-23 at 12 50 34 PM](https://github.com/prisma/prisma/assets/36806704/3e5b47a7-f5f6-42b8-8039-a25f355d00b4)

```sh
prisma generate
```

![Screenshot 2024-05-23 at 12 50 04 PM](https://github.com/prisma/prisma/assets/36806704/ae45307c-4538-40db-b0b7-ac7163a141f5)






I've also created a sister PR in the `withfig/autocomplete` repo: https://github.com/withfig/autocomplete/pull/2345

nickcarnival committed:
feat(cli): generate no hint argument Closes #22513

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/prisma/prisma?pullRequest=24274) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/prisma/prisma?pullRequest=24274) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/prisma/prisma?pullRequest=24274) it.</sub>

nickcarnival committed:
feat(cli): address failing tests

gutyerrez commented on packages/cli/src/Generate.ts:
```diff
@@ -48,12 +48,12 @@ ${bold('Usage')}
              --watch   Watch the Prisma schema and rerun after a change
          --generator   Generator to use (may be provided multiple times)
          --no-engine   Generate a client for use with Accelerate only
+        --no-hint     Hides the hint messages but still outputs errors and warnings
```
```suggestion
         --no-hint     Hides the hint messages but still outputs errors and warnings
```

nickcarnival replied:
56bde3bd4ca1f716cf6c337f73c7ce5ec39577b7

gutyerrez commented on packages/cli/src/Generate.ts:
```diff
@@ -104,6 +104,8 @@ ${bold('Examples')}
       '--data-proxy': Boolean,
       '--accelerate': Boolean,
       '--no-engine': Boolean,
+      // https://github.com/prisma/prisma/issues/22513
```
This is really relevant?

nickcarnival replied:
56bde3bd4ca1f716cf6c337f73c7ce5ec39577b7

nickcarnival committed:
addressed PR feedback

nickcarnival committed:
trigger CI again

nickcarnival committed:
consistent spelling of hints argument

nickcarnival commented:
@gutyerrez I've addressed your comments and the CI tests are now passing. It seems like someone has to add a milestone before the last check will pass but other than that it should be good to go :)

nickcarnival committed:
update message to trigger CI and run all tests rather than skipping some of them

nickcarnival committed:
feat: fix verbage

nickcarnival commented:
@Jolg42 I see that you removed the milestone. Is there something I need to do to get this added back?
It seems like that's the only check left for this PR.

nickcarnival committed:
(fix): pnpm run lint

nickcarnival commented on packages/cli/src/platform/auth/login.ts:
This change was made via `pnpm run lint`

nickcarnival commented on packages/client/tests/e2e/prisma-client-imports/src/default.ts:
This change was made via `pnpm run lint`

nickcarnival commented on packages/client/tests/functional/omit/test.ts:
This change was made via `pnpm run lint`

nickcarnival commented on packages/get-platform/src/test-utils/index.ts:
This change was made via `pnpm run lint`

nickcarnival commented:
@SevInf is there anything more I need to do to get this merged?

SevInf commented on packages/cli/src/Generate.ts:
```diff
@@ -154,6 +155,7 @@ ${bold('Examples')}
           Boolean(process.env.PRISMA_GENERATE_DATAPROXY) || // legacy, keep for backwards compatibility
           Boolean(process.env.PRISMA_GENERATE_ACCELERATE) || // legacy, keep for backwards compatibility
           Boolean(process.env.PRISMA_GENERATE_NO_ENGINE),
+        noHints: Boolean(args['--no-hints']),
```
Why forward this to generators? At no point this value actually get used right now.

nickcarnival replied:
It's gone now, thanks for catching this.

SevInf commented on packages/cli/src/Generate.ts:
```diff
@@ -309,6 +315,8 @@ More information: https://pris.ly/d/client`
 ${boxedTryAccelerateMessage}
 ${getHardcodedUrlWarning(config)}${breakingChangesStr}${versionsWarning}`
         }
+        const hideHintMode = args['--no-hints'] || false
```
Right now, the code still goes through all of the trouble of creating hint message just to throw it away at the last moment if `--no-hint` is set. Can we instead refactor the code so that those code paths would be completely skipped in `--no-hints` case?

nickcarnival replied:
I added another level of indentations and a large else block that accomplishes this 😆

SevInf commented on packages/cli/src/__tests__/commands/Generate.test.ts:
```diff
@@ -365,6 +365,60 @@ describe('using cli', () => {
     }
   })
 
+  it('should hide hints with --no-hints', async () => {
```
Thanks for adding the test!

SevInf commented on packages/cli/src/__tests__/commands/Generate.test.ts:
```diff
@@ -365,6 +365,60 @@ describe('using cli', () => {
+    `)
+    }
+
+    if (engineType !== ClientEngineType.Library && engineType !== ClientEngineType.Binary) {
```
This can never happen, engine type is always either binary or library. Where does this snapshot come from?

SevInf reviewed (changes requested):
Hi @nickcarnival!
Thank you for submitting the PR. I've left a few comments, after my feedback is addressed this should be good to merge.

nickcarnival committed:
(fix): skip unneded code paths and remove useless test

nickcarnival commented:
@SevInf thanks so much for the review, I believe I've addressed all of your comments.

SevInf committed:
Few stylistic changes

SevInf commented:
Thank you, waiting for final CI run and merging
